pkgdb2/api/admin.py

```diff
@@ -0,0 +1,355 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2014-2015  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions
+# of the GNU General Public License v.2, or (at your option) any later
+# version.  This program is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY expressed or implied, including the
+# implied warranties of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.  You
+# should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# Any Red Hat trademarks that are incorporated in the source
+# code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission
+# of Red Hat, Inc.
+#
+
+'''
+Admin interface for the API.
+'''
+
+import flask
+
+from math import ceil
+
+import pkgdb2.lib as pkgdblib
+import pkgdb2.forms
+from pkgdb2 import SESSION, is_admin
+from pkgdb2.api import API, get_limit
+
+
+@API.route('/admin/actions/')
+@API.route('/admin/actions')
+def api_admin_actions():
+    """
+List admin actions
+------------------
+    List actions requiring intervention from an admin.
+
+    ::
+
+        /api/admin/actions/
+
+    Accept GET queries only.
+
+    :kwarg package: restrict the actions to a specific package.
+    :kwarg packager: restrict the actions to a specific packager.
+    :kwarg action: restrict the actions to a specific action, options are:
+        ``request.branch``.
+    :kwarg status: restrict the actions depending on their status, options
+        are: ``Awaiting Review``, ``Approved``, ``Denied``, ``Obsolete``,
+        ``Removed``.
+    :kwarg limit: An integer to limit the number of results, defaults to
+        250, maximum is 500.
+    :kwarg page: The page number to return (useful in combination to limit).
+
+    Sample response:
+
+    ::
+
+        /api/admin/actions
+
+        {
+          "output": "ok",
+          "actions": [
+            {
+              "action": "request.branch",
+              "collection": {
+                "branchname": "epel7",
+                "dist_tag": ".el7",
+                "koji_name": "epel7",
+                "name": "Fedora EPEL",
+                "status": "Active",
+                "version": "7"
+              },
+              "date_created": 1402470695.0,
```

Comment:
That's not something I handle and I'd rather leave it the way python creates it :)